core/src/main/java/com/opentable/logging/JsonLogEncoder.java

```diff
@@ -88,7 +90,7 @@ public ObjectNode convertToObjectNode(ILoggingEvent event) {
         }
 
         for (Entry<String, String> e : event.getMDCPropertyMap().entrySet()) {
-            if (!logLine.has(e.getKey())) {
+            if (!logLine.has(e.getKey()) && HEADER_BLACKLIST.logFromMDC(e.getKey())) {
```

Comment:
I get how this logic works, but it is semantically confusing to go to the the _blacklist_ to determine what we are allowing. When reading this logic I keep thinking there is a missing `!`, even though I know the logic is correct.

Reply:
We chatted, open to changes. A negation must exist in either here or there ;)